scripts/schema/compare_schemas.js

```diff
@@ -0,0 +1,188 @@
+/* ADD_LICENSE_HEADER */
+/* eslint-disable no-console */
+"use strict"
+
+const { resolve } = require("path")
+
+// Compare trees recursively and print differences.
+const difftree = (a, b, deep) => {
+	const diffs = []
+	if (deep === undefined) {
+		deep = false
+	}
+
+	const remtree = (v, path) => {
+		if (v === undefined) {
+			return
+		}
+		if (!deep) {
+			diffs.push({ path: path, change: "-", val: v })
+		} else {
+			const sep = path === "" ? "" : "."
+			if (typeof v === "object") {
+				for (const key in v) {
+					remtree(v[key], path + sep + key)
+				}
+			} else {
+				diffs.push({ path: path, change: "-", val: v })
+			}
+		}
+	}
+
+	const addtree = (v, path) => {
+		if (v === undefined) {
+			return
+		}
+		if (!deep) {
```

Comment:
deep?

Reply:
Ahh, are those two functions otherwise similar but other adds and the other removes?

Reply:
Yes. I guess I could have just added +/- as a parameter instead of having two functions.